janus_server/src/aggregator/aggregate_share.rs

```diff
@@ -193,6 +193,11 @@ impl CollectJobDriver {
             .body(req.get_encoded())
             .send()
             .await?;
+        let status = response.status();
+        if !status.is_success() {
+            // TODO(#381): Attempt to decode a problem details response.
+            return Err(Error::Http(status));
+        }
```

Comment:
In cases like this where we treat anything in the 4xx-5xx range uniformly as an error, I think we could use [`reqwest::Response::error_for_status`](https://docs.rs/reqwest/latest/reqwest/struct.Response.html#method.error_for_status), which would save us a little code and should be handled by the existing `aggregator::Error::HttpClient` case which wraps `request::Error`.
```suggestion
            .await?
            .error_for_status()?;
```
Though maybe we shouldn't do this, because it's not apparent that `request::Error` lets you get at the response body, which we will eventually want to decode. Really what I'd like is for this to be harder to get wrong. Maybe we should write a wrapper around `reqwest::RequestBuilder` so we can do something like `RequestBuilder::send_with_expected_status(request, 201)`, and then you get back a `Result<Response, Error>`, which will be an error if the response status was anything but the expected HTTP status.

But we should just go ahead with your change for now.

Reply:
I think we are going to want to eventually attach different behaviors to different error types from the peer aggregator; at least, some errors will be immediately terminal, some will allow a retry, based on the content of the problem document. (This is tracked in #235.)

That said, I think any abstraction we build should capture that ability (or at least not need to be unwound once we decide to build out #235).